src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private int size;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+
+    private void throwArrayListIndexOutOfBoundsException() {
+        throw new ArrayListIndexOutOfBoundsException("Array List Index Out Of Bounds Exception");
+
+    }
+
+    private void resizeArray() {
+        final int newCapacity = elementData.length + elementData.length / 2;
+        T[] newArray = (T[]) new Object[newCapacity];
+        System.arraycopy(elementData, 0, newArray, 0, size);
+        elementData = newArray;
+    }
+
+    private void shiftElementsToRight(int startIndex) {
+        for (int i = size - 1; i >= startIndex; i--) {
+            elementData[i + 1] = elementData[i];
+        }
+    }
+
+    private void shiftElementsToLeft(int startIndex) {
+        for (int i = startIndex; i < size - 1; i++) {
+            elementData[i] = elementData[i + 1];
+        }
+        elementData[size - 1] = null;
+    }
```

Comment:
you can use System.arrayCopy to shift vlaues, no need to create another array - just copy values to the same array